app/client/components/register/AccomodationFieldGroup.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useReducer } from "react";
+import { WidgetHandler } from "reactivated/dist/forms";
+import { DjangoFormsWidgetsCheckboxInput } from "reactivated/dist/generated";
+import { BasicFormField } from "../forms/BasicFormField";
+
+interface AccomodationFieldGroupProps {
+  dinnerField: WidgetHandler<DjangoFormsWidgetsCheckboxInput>;
+  accomodationField: WidgetHandler<DjangoFormsWidgetsCheckboxInput>;
+  breakfastField: WidgetHandler<DjangoFormsWidgetsCheckboxInput>;
+}
+
+interface GroupState {
+  accomodation: {
+    checked: boolean;
+  };
+  breakfast: {
+    disabled: boolean;
+    checked: boolean;
+  };
+  dinner: {
+    disabled: boolean;
+    checked: boolean;
+  };
+}
+
+interface GroupAction {
+  fieldName: string;
+  newValue: boolean;
+}
+
+export const AccomodationFieldGroup = ({
+  dinnerField,
+  accomodationField,
+  breakfastField,
+}: AccomodationFieldGroupProps) => {
+  const reducer = (state: GroupState, action: GroupAction): GroupState => {
+    const { fieldName, newValue } = action;
+
+    switch (fieldName) {
+      case dinnerField.name:
+        return {
+          ...state,
+          dinner: {
+            ...state.dinner,
+            checked: newValue,
+          },
+        };
+
+      case accomodationField.name:
+        return {
+          accomodation: {
+            checked: newValue,
+          },
+          breakfast: {
+            disabled: !newValue,
+            checked: newValue,
+          },
+          dinner: {
+            disabled: !newValue,
+            checked: newValue,
+          },
+        };
+
+      case breakfastField.name:
+        return {
+          ...state,
+          breakfast: {
+            ...state.breakfast,
+            checked: newValue,
+          },
+        };
+      default:
+        return state;
+    }
+  };
+
+  const [groupState, dispatch] = useReducer(reducer, {
+    accomodation: {
+      checked: accomodationField.value,
+    },
+    breakfast: {
+      disabled: breakfastField.disabled || !accomodationField.value,
+      checked: breakfastField.value,
+    },
+    dinner: {
+      disabled: dinnerField.disabled || !accomodationField.value,
+      checked: dinnerField.value,
+    },
+  });
+
+  useEffect(() => {
```

Comment:
I don't like this. If we need to keep it synchronized with Reactivated form state, we must change this logic to use those handlers. Currently, we are doing unnecessary rerender just to keep the state synchronized when we can directly operate on main state.

Reply:
Done, it actually made the code much simpler.

I was struggling to figure out how to work with Reactivated's form state and I think this is the way, thanks for this comment :)